src/devices/amina.ts

```diff
@@ -0,0 +1,357 @@
+import {Zcl} from 'zigbee-herdsman';
+
+import * as constants from '../lib/constants';
+import * as exposes from '../lib/exposes';
+import {binary, deviceAddCustomCluster, electricityMeter, numeric, onOff} from '../lib/modernExtend';
+import * as ota from '../lib/ota';
+import * as reporting from '../lib/reporting';
+import {DefinitionWithExtend, Fz, KeyValue, Tz} from '../lib/types';
+import * as utils from '../lib/utils';
+
+const e = exposes.presets;
+const ea = exposes.access;
+
+const manufacturerOptions = {manufacturerCode: 0x143b};
+
+const aminaControlAttributes = {
+    cluster: 0xfee7,
+    alarms: 0x02,
+    ev_status: 0x03,
+    connect_status: 0x04,
+    single_phase: 0x05,
+    offline_current: 0x06,
+    offline_single_phase: 0x07,
+    time_to_offline: 0x08,
+    enable_offline: 0x09,
+    total_active_energy: 0x10,
+    last_session_energy: 0x11,
+};
+
+const aminaAlarms = [
+    'Welded relay(s)',
```

Comment:
Please make all of these `snake_case`:

```suggestion
    'welded_relays',
```

Reply:
Just curious, what is the purpose of converting those to snake case but not status texts for ev_status for instance?

Reply:
Usually text is free format, but for this use case it may be better to use a `e.enum()`

Reply:
Good point, `e.list()` with an `e.enum()` as the type would be the best then.

Reply:
Do you have any examples of how to do this, because whatever I do, the GUI only shows an array of strings, or is this expected behaviour?
I defined:

```typescript
const aminaAlarms = [
    'no_alarm',
    'welded_relay', // Bit 0
    'wrong_voltage_balance',
    'rdc_dd_dc_leakage',
    'rdc_dd_ac_leakage',
    'high_temperature',
    'overvoltage',
    'undervoltage',
    'overcurrent',
    'car_communication_error',
    'charger_processing_error',
    'critical_overcurrent',
    'critical_powerloss',
    'unknown_alarm_bit_12',
    'unknown_alarm_bit_13',
    'unknown_alarm_bit_14',
    'unknown_alarm_bit_15',
];

const aminaAlarmsEnum = e.enum('alarms', ea.STATE_GET, aminaAlarms);
``` 

In the fzLocal.alarms converter:
```typescript
if (msg.data.alarms !== undefined) {
    const activeAlarms = [];
    result.alarm_active = false;

    for (let i = 0; i < 16; i++) {
        if ((msg.data['alarms'] >> i) & 0x01) {
            activeAlarms.push(aminaAlarmsEnum.values[i+1]);
            result.alarm_active = true;
        }
    }

    if (result.alarm_active === false) {
        activeAlarms.push(aminaAlarmsEnum.values[0]);
    }

    result.alarms = activeAlarms;
    return result;
}
``` 

And finally the expose:
```typescript
e.list('alarms', ea.STATE_GET, aminaAlarmsEnum).withDescription('List of active alarms'),
``` 

Example result with bit 1, 3 and 12 enabled:
![image](https://github.com/user-attachments/assets/124695fe-d5e9-4655-9944-30bf428e5a53)

Example result with no alarms:
![image](https://github.com/user-attachments/assets/e1069607-855d-466d-9b5b-94dc4b63bf26)

I would perhaps expect something like this, not editable of course:
![image](https://github.com/user-attachments/assets/7113e4ce-ee1f-4607-a120-1289deaa3be3)

Reply:
This is expected indeed, the frontend should be improved a bit here

Reply:
Alright then, this is implemented in https://github.com/Koenkk/zigbee-herdsman-converters/pull/8191/commits/8263316e36234ac8b0e44e533972829ff056690b